Lib/traceback.py

```diff
@@ -1095,6 +1095,26 @@ def __init__(self, exc_type, exc_value, exc_traceback, *, limit=None,
                         self._str += f" Or did you forget to import '{wrong_name}'?"
                     else:
                         self._str += f". Did you forget to import '{wrong_name}'?"
+        elif exc_value and self._is_unpack_error(exc_value):
```

Comment:
I find sticking the secret attributes a bit flaky and I would have preferred to raise an entire different subclass for unpacking errors but if we want to continue with this approach, I would recommend to add more checks to `_is_unpack_error` to protect this code path in the future: check that both attributes are present and that the expected types are met.